Sources/Xcodeproj/XcodeProjectModelSerialization.swift

```diff
@@ -470,19 +470,19 @@ fileprivate class PropertyListSerializer {
     /// dictionary is what gets written to the property list.
     var idsToDicts = [String: PropertyList]()
 
-    /// Returns the identifier for the object, assigning one if needed.
+    /// Returns the quoted identifier for the object, assigning one if needed.
     func id(of object: PropertyListSerializable) -> String {
         // We need a "serialized object ref" wrapper for the `objsToIds` map.
         let serObjRef = SerializedObjectRef(object)
         if let id = objsToIds[serObjRef] {
-            return id
+            return "\"\(id)\""
```

Comment:
This won't work if the id contains quotes or escapes... really, the quoting/escaping should happen at the point at which this is rendered to a file, but that can be fixed in a follow-on PR.